Tests/MapboxSearchIntegrationTests/search-box/SearchBox_SearchEngineIntegrationTests.swift

```diff
@@ -0,0 +1,197 @@
+        wait(for: [updateExpectation], timeout: 10)
+        XCTAssertFalse(searchEngine.suggestions.isEmpty)
+
+        let errorExpectation = delegate.errorExpectation
+        let selectedResult = try XCTUnwrap(searchEngine.suggestions.first)
+        searchEngine.select(suggestion: selectedResult)
+        wait(for: [errorExpectation], timeout: 10)
+
+        if case .resultResolutionFailed(let suggestion) = delegate.error {
+            XCTAssertEqual(selectedResult.id, suggestion.id)
+            XCTAssertEqual(selectedResult.name, suggestion.name)
+            XCTAssertEqual(selectedResult.address, suggestion.address)
+        } else {
+            XCTFail("Not expected")
+        }
+
+        XCTAssertNil(delegate.resolvedResult)
+    }
+
+    func testBatchResolve() throws {
+        try server.setResponse(.multiRetrieve)
+
+        let results = CoreSearchResultStub.makeMixedResultsSet().map(\.asCoreSearchResult)
+        let coreResponse = CoreSearchResponseStub.successSample(results: results)
+
+        let suggestions = CoreSearchResultStub.makeSuggestionsSet().map(\.asCoreSearchResult).compactMap {
+            SearchResultSuggestionImpl(coreResult: $0, response: coreResponse)
+        }
+
+        let expectation = delegate.batchUpdateExpectation
+        searchEngine.select(suggestions: suggestions)
+
+        wait(for: [expectation], timeout: 10)
+        XCTAssertFalse(delegate.resolvedResults.isEmpty)
+    }
+
+    func testBatchResolveFailed() throws {
+        try server.setResponse(.multiRetrieve, statusCode: 500)
+
+        let results = CoreSearchResultStub.makeMixedResultsSet().map(\.asCoreSearchResult)
+        let coreResponse = CoreSearchResponseStub.successSample(results: results)
+
+        let suggestions = CoreSearchResultStub.makeSuggestionsSet().map(\.asCoreSearchResult).compactMap {
+            SearchResultSuggestionImpl(coreResult: $0, response: coreResponse)
+        }
+
+        let expectation = delegate.errorExpectation
+        searchEngine.select(suggestions: suggestions)
+
+        wait(for: [expectation], timeout: 10)
+        XCTAssert(delegate.error?.errorCode == 500)
+    }
+
+    func testSuggestionTypeQuery() throws {
+        try server.setResponse(.recursion)
+
+        let updateExpectation = delegate.updateExpectation
+        searchEngine.query = "Recursion"
+        wait(for: [updateExpectation], timeout: 10)
+
+        let firstSuggestion = try XCTUnwrap(searchEngine.suggestions.first)
+        searchEngine.select(suggestion: firstSuggestion)
+
+        /// Mock that a subsequent retrieve call has been made and succeeds.
+        /// This is not implemented in the stub delegate (it reports success).
+        /// We have validated that the chained operations are connected.
+        let nextUpdateExpectation = delegate.updateExpectation
+        wait(for: [nextUpdateExpectation], timeout: 10)
+        XCTAssertFalse(searchEngine.suggestions.isEmpty)
```

Comment:
@azarovalex this test function is passing locally but failing on CircleCI, does anything stand-out as wrong?

Reply:
Hmm I've just tried running these tests in Xcode as well as `make ci-full-test` and it failed both times.
Could it be that we can't wait for an expectation that was already fulfilled before? Or maybe `expectedFulfillmentCount` to 2.

Also some other tests from this file fail for me on aa1b605e6f1a84ae69cd8cfb8a3c76251bf99117
<img width="426" alt="image" src="https://github.com/mapbox/mapbox-search-ios/assets/1976216/4dd25914-90f2-407e-b29a-3f6ed1189a2b">

Reply:
Those other tests will be fixed in SAPI-1176